docs/source/models/segment_anything.rst

```diff
@@ -0,0 +1,251 @@
+    )
+
+    # Runs the prediction with all prompts
+    prediction = predictor(
+        image=image,
+        point_coords=input_point,
+        point_labels=input_label,
+        boxes=input_box,
+        multimask_output=True,
+    )
+
+    #----------------------------------------------
+    # or run the prediction with just the keypoints
+    prediction = predictor(
+        image=image,
+        point_coords=input_point,
+        point_labels=input_label,
+        multimask_output=True,
+    )
+
+    #----------------------------------------------
+    # or run the prediction with just the box
+    prediction = predictor(
+        image=image,
+        boxes=input_box,
+        multimask_output=True,
+    )
+
+    #----------------------------------------------
+    # or run the prediction without prompts
+    prediction = predictor(
+        image=image,
+        multimask_output=True,
+    )
+
+    #------------------------------------------------
+    # or run the prediction using the previous logits
+    prediction = predictor(
+        image=image,
+        mask_input=prediction.logits
+        multimask_output=True,
+    )
+
+    # The `prediction` is a dataclass with the masks, scores and logits
+    print(prediction.masks.shape)
+    print(prediction.scores)
+    print(prediction.logits.shape)
+
+.. Mask Generator
+.. ^^^^^^^^^^^^^^
+
+
+Example of how to use the SAM model without API
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This is a simple example, of how to directly use the SAM model loaded. We recommend the use of
+:ref:`Predictor API<anchor Predictor>` to handle/prepare the inputs.
+
+.. code-block:: python
+
+    from kornia.contrib import Sam
+    from kornia.contrib.sam.base import SegmentationResults
+    from kornia.io import load_image, ImageLoadType
+    from kornia.utils import get_cuda_device_if_available
+    from kornia.geometry import resize
+
+    model_type = 'vit_b' # or can be a number `2` or the enum sam.SamModelType.vit_b
+    checkpoint_path = './path_for_the_model_checkpoint.pth'
+    device = get_cuda_device_if_available()
+
+    # Load the model
+    sam_model = Sam.from_pretrained(model_type, checkpoint_path, device)
+
+    # Load image
+    image = load_image('./example.jpg', ImageLoadType.RGB8, device).float()
+
+    # Transform the image (CxHxW) into a batched input (BxCxHxW)
+    image = image[None, ...]
+
+    # Resize the image to have the maximum size 1024 on its largest side
+    inpt = resize(image, 1024, side='long')
+
+    # Embed prompts -- ATTENTION: should match the coordinates after the resize of the image
+    sparse_embeddings, dense_embeddings = sam_model.prompt_encoder(points=None, boxes=None, masks=None)
+
+    # Preprocess input
+    inpt = (inpt - sam_model.pixel_mean) / sam_model.pixel_std
+    padh = model_sam.image_encoder.img_size - h
+    padw = model_sam.image_encoder.img_size - w
+    inpt = pad(inpt, (0, padw, 0, padh))
```

Comment:
This makes me thing to have a pad_image with the following signature 
```python
def pad_image(image: Tensor | Image, tuple[int, int] | ImageSize) -> Tensor | Image:
    pass
```

Reply:
i am not sure where, but I think we had something similar to this, as "pad right bottom"